src/relay.cpp

```diff
@@ -534,10 +618,28 @@ void relay_relay_forw(int sock, const uint8_t *msg, int32_t len, const ip6_hdr *
     memcpy(current_buffer_position, &new_message, sizeof(dhcpv6_relay_msg));
     current_buffer_position += sizeof(dhcpv6_relay_msg);
 
+    // insert option82 for new relay-forward packet, we need this information
+    // to get original relay-forward source interface for accurate counting in dualtor env
+    if (dual_tor_sock) {
```

Comment:
I'd suggest using` if (config->is_interface_id) ` check here instead of dual_tor_sock. Also set the is_interface_id to True in processRelayNotification by default for DualTor. 

Also, the logic of adding Option82 is duplicated in relay_relay_forw & relay_client. If possible, refactor into a separate function.

Reply:
Good idea, this seems flexible compare with current behavior. Will apply in next diff.

Reply:
Applied in latest diff.